File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int low=0,high=n-1;
        int maxiarea=0;
        while(low<high)
        {
            int maxheight=min(height[low],height[high]);
             maxiarea=max(maxiarea,(high-low)*maxheight);
            if(height[low]<height[high])
            {
                low++;
            }
            else
            {
                high--;
            }
        }
        return maxiarea;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] nums) {
        int max=0;
        
        int low=0;
        int high=nums.length-1;

        while(low<high)
        {
            int min=Math.min(nums[low],nums[high]);

            max=Math.max(max,(high-low)*min);

            if(nums[low]>nums[high])
            {
                high--;
            }
            else
            {
                low++;
            }
            
        }

        return max;
    }
}